### PR TITLE
Add support for testdouble-chai

### LINKED ISSUE #19: Support for testdouble-chai
similar to #14

see https://github.com/BaseCase/testdouble-chai

this would deprecate https://github.com/alexlafroscia/ember-cli-testdouble-chai

also note that this might conflict with `sinon-chai`, so we should make sure that only one of them can be used at the same time.

### PR DESCRIPTION
**Pull request by alexlafroscia**
## TODO

- [x] Figure out how to test both `testdouble-chai` and `sinon-chai`, when both can't be enabled
- [x] Test in dummy app

Closes #19

### CODE REVIEW

**Comment by alexlafroscia** [comment]
Thinking about the fact that they conflict... Do you think it would be possible to patch one/both of `sinon-` and `testdouble-chai` so that they support both being enabled? It feels like something that should be possible, but I don't know a ton about how either of them work.

**Comment by Turbo87** [comment]
@alexlafroscia I've asked a very similar question in https://github.com/nathanboktae/chai-dom/issues/16 and we've concluded that it doesn't seem possible with the current API that Chai has for addons.

As you can see in the lack of tests for `chai-dom` I haven't yet figured out how to test these two either. If you have some ideas on how to do it I'd like to hear them. For the time being we should test those mutually exclusive plugins manually and try to figure out a way to automate it.

Can you adjust the PR to only add support for the testdouble stuff for now but not include the failing test?

**Comment by alexlafroscia** [comment]
Yup, absolutely.

File: index.js [hunk 1]
```diff
@@ -49,12 +49,19 @@ var sinonPlugin = {
   path: 'sinon-chai.js',
 };
 
+var testdoublePlugin = {
+  name: 'testdouble-chai',
+  constraint: '^0.5.0',
```
Comment (Turbo87) [inline index.js]: what about the releases below `0.5.0`? they don't work?

**Comment by alexlafroscia** [comment]
I think I might need to add some code here to handle bootstrapping `testdouble-chai` 

https://github.com/BaseCase/testdouble-chai#setup

`ember-cli-testdouble-chai` adds to the `test-helper.js` file, but it would be great to avoid the need for that entirely

**Comment by Turbo87** [comment]
indeed, their UMD wrapper doesn't seem to call `chai.use()` automatically. I had to do something similar for `chai-as-promised`, but that also included using rollup and other things which might not be necessary here.

**Comment by alexlafroscia** [comment]
Yeah, I'm thinking we can just include a file in the addon's `vendor` folder that gets pulled in when `testdouble-chai` is present.

**Comment by alexlafroscia** [comment]
Just found out about the [`overwriteMethod`](http://chaijs.com/api/plugins/#method_overwriteproperty) option that plugins can use -- I'm surprised that doesn't work for this sinon/testdouble case.

**Comment by Turbo87** [comment]
Yeah but you can only overwrite existing methods, if you overwrite an unexisting method it won't work :-/

**Comment by alexlafroscia** [comment]
Not sure if you had something else in mind, but I ended up adding an additional, option property to the `plugin` format that can specify an additional file that needs to be loaded. I made a dummy app that was linked to my local version of `ember-cli-chai` and was able to get everything working in there through this setup.

**Comment by Turbo87** [comment]
released as v0.3.2 👍

**Comment by alexlafroscia** [comment]
As a note, I added a notice to the `ember-cli-testdouble-chai` README that mentions the fact that it's now deprecated.

**Pull request merged**